### PR TITLE
Add DotNetHowToAutoComplete sample solution

### PR DESCRIPTION
**Pull request by mhko**
@brjohnstmsft, @Yahnoosh 

This is the dot net sample for autocomplete.

### CODE REVIEW

File: DotNetHowToAutoComplete/DotNetHowToAutoComplete/packages.config [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Microsoft.Azure.Search.Common" version="5.0.0" targetFramework="net461" />
```
Comment (bitsforthought) [inline DotNetHowToAutoComplete/DotNetHowToAutoComplete/packages.config:3]: You'll need to reference version 6-preview in order for this to work. You can test it locally by running `Build-SeachPackages.ps1`, and then make sure to include the binaries/packages directory in your local azure-sdk-for-net repo in your NuGet package sources.

**Review by bitsforthought — CHANGES_REQUESTED** [review]
This sample is consistent with our other samples, but is it useful? Since this feature is going to be used for type-ahead, perhaps we should have a sample with more sophisticated UX than a console app?

@HeidiSteen what do you think?

**Comment by HeidiSteen** [comment]
Mike Carter said he was trying to figure out how to do a bigger sample/tutorial around this.  Have you talked to him?  He is targeting autocomplete, and I believe he is also planning on contrasting autocomplete with suggesters, maybe doing some kind of side-by-side demo not unlike this one (for analyzers) : http://alice.unearth.ai/

**Comment by bitsforthought** [comment]
@HeidiSteen that's a good suggestion. Assuming Mike creates a more comprehensive article/sample around Autocomplete, do you think we should still publish this sample? Does it provide enough value in your opinion?

@mhko -- Can you please follow up with Mike?

**Comment by mhko** [comment]
Thanks @HeidiSteen @brjohnstmsft! Mike did talk to me about creating a post that covers both suggest and autocomplete. I am unsure if he also plans to write a sample. Will follow up.

**Comment by bitsforthought** [comment]
Closing this since we've decided to work on a sample that isn't just a console app.

**Pull request closed without merging**